### PR TITLE
feat(tag-group): create new component

### PR DESCRIPTION
**Pull request by IgnacioBecerra**
### Related Ticket(s)
#5996

### Description
Create new TagGroup Component in both Web Components and React Wrapper. This also includes Sandbox examples.

Screen Shot 2021-05-17 at 8 43 37 AM

### Changelog

**New**

- `TagGroup` component for WC and React Wrapper

### CODE REVIEW

**Review by shixiedesign — COMMENTED** [review]
1. Awesome but NO combining 😂 
image

2. Please remove `Tag link (href)` knob. 
image

3. Is it possible to have a `Tag type` knob that's a dropdown, allowing users to pick between Tag link, Carbon tags? 

4. Also didn't see the Tag group in React wrapper. Not sure if it's forgotten or out of scope?

File: packages/web-components/src/components/tag-group/__stories__/README.stories.mdx [hunk 1]
```diff
@@ -0,0 +1,61 @@
+import { Props, Description } from '@storybook/addon-docs/blocks';
+import contributing from '../../../../../../docs/contributing-license.md';
+import markdownContents from '../../../../docs/markdown-contents.md';
+
+# Tag Link
```
Comment (ariellalgilmore) [inline packages/web-components/src/components/tag-group/__stories__/README.stories.mdx:5]:
```suggestion
# Tag Group
```

File: packages/web-components/src/components/tag-group/__stories__/README.stories.react.mdx [hunk 2]
```diff
@@ -0,0 +1,44 @@
+import { Preview, Props, Description, Story } from '@storybook/addon-docs/blocks';
+import contributing from '../../../../../../docs/contributing-license.md';
+import { PropTypesRef } from '@carbon/ibmdotcom-web-components/es/components-react/tag-group/tag-group';
+
+# Tag Link
```
Comment (ariellalgilmore) [inline packages/web-components/src/components/tag-group/__stories__/README.stories.react.mdx:5]:
```suggestion
# Tag Group
```

File: packages/web-components/src/components/tag-group/__stories__/README.stories.react.mdx [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+
+```javascript
+import DDSTagGroup from '@carbon/ibmdotcom-web-components/es/components-react/tag-group/tag-group.js';
+import DDSTagLink from '@carbon/ibmdotcom-web-components/es/components-react/tag-link/tag-link.js';
+
+function App() {
+  return (
+  <dds-tag-link href="https://example.com">
+    Brand: Watson
+  </dds-tag-link>
```
Comment (ariellalgilmore) [inline packages/web-components/src/components/tag-group/__stories__/README.stories.react.mdx:27]:
```suggestion
<dds-tag-group>
  <dds-tag-link href="https://example.com">
    Brand: Watson
  </dds-tag-link>
  <dds-tag-link href="https://example.com">
    Brand: Watson
  </dds-tag-link>
</dds-tag-group>
```

File: packages/web-components/src/components/tag-group/__stories__/README.stories.react.mdx [hunk 4]
```diff
@@ -0,0 +1,44 @@
...
+
+<Props of={PropTypesRef} />
+
+## Stable selectors
+
+See [our README](https://github.com/carbon-design-system/carbon-for-ibm-dotcom/tree/master/packages/web-components#stable-selectors-for-analytics-and-integratione2e-testing-in-web-components) to see how Web Components selector and `data-autoid` should be used.
+
+| Web Components selector | Compatibility selector | Description |
+| ----------------------- | ---------------------- | ----------- |
+| `<dds-tag-link>`        | N/A                    | Component   |
```
Comment (ariellalgilmore) [inline packages/web-components/src/components/tag-group/__stories__/README.stories.react.mdx:42]:
```suggestion
| `<dds-tag-group>`        | N/A                    | Component   |
```

**Comment by shixiedesign** [comment]
Web components look good! React wrapper showing a bit of an issue when the dropdown selects Carbon tag: 
May-18-2021 21-44-53

**Review by shixiedesign — APPROVED** [review]
Expressive theme might be the reason Carbon sizing is off. Discount on tag link in group delivery this is done!

**Review by ariellalgilmore — APPROVED** [review]
LGTM! Thanks @IgnacioBecerra !

**Pull request merged**